app/Booking.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App;
+
+use App\Exceptions\OverlappingBookingException;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Support\Facades\DB;
+
+class Booking extends Model
+{
+    protected $fillable = ['user_id', 'position_id', 'from', 'to'];
+
+    public function position(): BelongsTo
+    {
+        return $this->belongsTo(Position::class);
+    }
+
+    public static function canBeMade($position_id, $from, $to)
+    {
+        return !self::where('position_id', $position_id)
+            ->where(function ($query) use ($from, $to){
```

Comment:
Could this second where clause be put into a subQuery?

REF: https://medium.com/@joelbutcher/using-sub-queries-in-laravel-260b72be6c8

Also, see a talk by Jonathan Reinink: https://www.youtube.com/watch?v=IBUXXErAtuk